app/src/main/java/ca/marshallasch/veil/comparators/PostComparator.java

```diff
@@ -0,0 +1,25 @@
+package ca.marshallasch.veil.comparators;
+
+import java.util.Comparator;
+
+import ca.marshallasch.veil.proto.DhtProto;
+import ca.marshallasch.veil.utilities.Util;
+
+
+/**
+ * This class is used to sort {@link DhtProto.Post} objects by the time that they were created.
+ * This will sort it in decreasing order, (newest first)
+ * @author Marshall Asch
+ * @version 1.0
+ * @since 2018-06-13
+ */
+public class PostComparator implements Comparator<DhtProto.Post>
+{
+    @Override
+    public int compare(DhtProto.Post a, DhtProto.Post b)
```

Comment:
Is this sifting through every post 2 at a time and sorting it? Because if so, then we might have a problem.

Reply:
no this is only a comparator class, it is used to determine how to sort things. its how things are normally compared.